RoslynPluginGenerator/ZipExtensions.cs

```diff
@@ -58,4 +60,32 @@ public static void CreateFromDirectory(string sourceDirectoryName, string destin
             }
         }
     }
+
+    /// <summary>
+    /// Replaces the back slash to foward slash to attend the international convention of directories separation.
+    /// This will allow the zip file to be unzipped the correct way both linux and windows OS's.
+    /// </summary>
+    /// <seealso cref="System.Text.UTF8Encoding" />
+    class PathEncoder : UTF8Encoding
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PathEncoder"/> class.
+        /// </summary>
+        public PathEncoder() : base(true)
```

Comment:
In case we need explicit value for this parameter, we should use the overload with two parameters and throw on invalid characters, as per MS doc's recommendations.